driver/ppm_fillers.c

```diff
@@ -152,6 +152,44 @@ static inline struct pid_namespace *pid_ns_for_children(struct task_struct *task
 }
 #endif /* UDIG */
 
+
+/*
+ * Detect whether the file being referenced is an anonymous file created using memfd_create()
+ * and is being executed by referencing its file descriptor (fd). This type of file does not
+ * exist on disk and resides solely in memory, but it is treated as a legitimate file with an
+ * inode object and other file attributes.
+ *
+ **/
+static inline uint32_t get_exe_from_memfd(const struct file *exe_file)
+{
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 17, 0)
+	if(!(exe_file &&
+		 exe_file->f_path.dentry &&
+		 exe_file->f_path.dentry == exe_file->f_path.dentry->d_parent))
+	{
+		return 0;
+	}
+
+	int i = 0;
+	const char expected_prefix[] = "memfd:";
+	const char *fname = exe_file->f_path.dentry->d_name.name;
+
+	for(i = 0; fname && i < sizeof(expected_prefix); i++)
+	{
+		if(expected_prefix[i] != fname[i])
+		{
+			break;
+		}
+		else if(expected_prefix[i] == ':')
+		{
+			return PPM_EXE_FROM_MEMFD;
+		}
+	}
```

Comment:
Two things:

1. This is not eBPF, I'd assume we can use strncmp or something?
2. Is there really no other way to detect memfds? Something like superblock magic of the inode?

Reply:
1. This is more to keep it consistent for the reader wrt eBPF.
2. Not one I know of. This is the only way I could figure out.

Reply:
1. If we're going for consistency, I'd rather have a macro for all engines with potentially different definitions. It's not immediately obvious this is open-coded strncmp.
2. :sadpanda: